File: 2022/01/01.ts
//const line = require('fs').readFileSync('example.txt').toString().split('\n');
const line = require('fs').readFileSync('input.txt').toString().split('\n');

function getCalories() {
    let foodCount = new Array()
    let elf = 0
    line.forEach((cal, idx, array) => {
        if (cal !== '') elf += parseInt(cal)

        if (cal === '' || idx === array.length - 1) {
            foodCount.push(elf)
            elf = 0
            return;
        }

    });
    return foodCount.sort((a,b)=>b-a)[0]
}

console.log('Answer: ' + getCalories())

File: 2024/day01/01.py
left = []
right = []

with open('input.txt', 'r') as file:
#with open('example.txt', 'r') as file:
    for line in file:
        left.append(int(line.split()[0]))
        right.append(int(line.split()[1]))

left.sort()
right.sort()

total = 0
for i in range(len(right)):
    total += right[i] - left[i]
    
print(total)